lib/DWARFImporter/DWARFImporter.cpp

```diff
@@ -0,0 +1,189 @@
+
+  ~DWARFModuleUnit() = default;
+
+public:
+  DWARFModuleUnit(ModuleDecl &M)
+      : LoadedFile(FileUnitKind::DWARFModule, M) {}
+
+  virtual bool isSystemModule() const override { return false; }
+
+  virtual void
+  lookupValue(ModuleDecl::AccessPathTy accessPath, DeclName name,
+              NLKind lookupKind,
+              SmallVectorImpl<ValueDecl *> &results) const override {}
+
+  virtual TypeDecl *
+  lookupNestedType(Identifier name,
+                   const NominalTypeDecl *baseType) const override {
+    return nullptr;
+  }
+
+  virtual void lookupVisibleDecls(ModuleDecl::AccessPathTy accessPath,
+                                  VisibleDeclConsumer &consumer,
+                                  NLKind lookupKind) const override {}
+
+  virtual void
+  lookupClassMembers(ModuleDecl::AccessPathTy accessPath,
+                     VisibleDeclConsumer &consumer) const override {}
+
+  virtual void
+  lookupClassMember(ModuleDecl::AccessPathTy accessPath, DeclName name,
+                    SmallVectorImpl<ValueDecl *> &decls) const override {}
+
+  void lookupObjCMethods(
+      ObjCSelector selector,
+      SmallVectorImpl<AbstractFunctionDecl *> &results) const override {}
+
+  virtual void
+  getTopLevelDecls(SmallVectorImpl<Decl *> &results) const override {}
+
+  virtual void
+  getDisplayDecls(SmallVectorImpl<Decl *> &results) const override {}
+
+  virtual void
+  getImportedModules(SmallVectorImpl<ModuleDecl::ImportedModule> &imports,
+                     ModuleDecl::ImportFilter filter) const override {}
+
+  virtual void getImportedModulesForLookup(
+      SmallVectorImpl<ModuleDecl::ImportedModule> &imports) const override {};
+
+  virtual void collectLinkLibraries(
+      ModuleDecl::LinkLibraryCallback callback) const override {};
+
+  Identifier
+  getDiscriminatorForPrivateValue(const ValueDecl *D) const override {
+    llvm_unreachable("no private decls in Clang modules");
+  }
+
+  virtual StringRef getFilename() const override { return ""; }
+
+  virtual const clang::Module *getUnderlyingClangModule() const override {
+    return nullptr;
+  }
+
+  static bool classof(const FileUnit *file) {
+    return file->getKind() == FileUnitKind::DWARFModule;
+  }
+  static bool classof(const DeclContext *DC) {
+    return isa<FileUnit>(DC) && classof(cast<FileUnit>(DC));
+  }
+};
+
+class DWARFImporter::Implementation {
+public:
+  ASTContext &SwiftContext;
+  clang::ASTContext *ClangASTContext = nullptr;
+  clang::CompilerInstance *Instance = nullptr;
+  clang::Preprocessor *PP = nullptr;
+  clang::Sema *Sema = nullptr;
+
+  llvm::DenseMap<Identifier, DWARFModuleUnit *> ModuleWrappers;
+  Implementation(ASTContext &ctx, const ClangImporterOptions &clangImporterOpts)
+      : SwiftContext(ctx) {}
+
+  ModuleDecl *loadModule(SourceLoc importLoc,
+                         ArrayRef<std::pair<Identifier, SourceLoc>> path) {
+    Identifier name = // SwiftContext.getIdentifier((*clangModule).Name);
+        path[0].first;
```

Comment:
Not sure what the comment is supposed to mean. A module loader is always given a name to look up, not an already-extent module to load.

Reply:
That is the commented out code from the real ClangImporter. It should read: 
```
// FIXME: Implement submodule support.
```